common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -86,6 +87,52 @@ def default(self, obj):
         return json.JSONEncoder.default(self, obj)
 
 
+class WarningFieldSingleValueMixin(JSONField):
+    """
+    Wraps around the actual value of another xmodule.field.Field.
+    This mixin must be used with a field that does not hard-code
+    a set of values, as it uses the "_values" field to communicate
+    warnings to the front-end
+
+    The condition should be of the form "<fieldname> <op> <value>"
+    similar to a SQL where clause. Currently supported "op"
+    is only ">", and can be implemented in a need-by-need basis
+    on the front-end. "fieldname" is a CapaField.
+    """
+    def __init__(self, warning=None, warning_condition=None, **kwargs):
+        super(WarningFieldSingleValueMixin, self).__init__(**kwargs)
+        self._values = {
+            "original_value": self._values,
```

Comment:
actually, this doesn't seem quite consistent with the definition of `values` here: https://github.com/edx/XBlock/blob/master/xblock/fields.py#L280